src/routes/MemberProfile/MemberProfile.connector.js

```diff
@@ -41,14 +41,12 @@ export function mapStateToProps (state, props) {
   }
 }
 
-export function mapDispatchToProps (dispatch, props) {
-  const { routeParams, querystringParams } = props
-
+export function mapDispatchToProps (dispatch) {
   return {
     fetchPerson: (id) => dispatch(fetchPerson(id)),
     blockUser: (id) => dispatch(blockUser(id)),
     push,
-    showDetails: (id) => dispatch(push(postUrl(id, routeParams, querystringParams)))
+    showDetails: (id, routeParams) => dispatch(push(postUrl(id, routeParams)))
```

Comment:
it feels weird to pass in the routeParams here, i dont quite get it...

Reply:
The `util/navigation` helpers were designed to be coupled to the react router key schema in this way. We could de-couple them but in my experience we just end-up modeling or repeating the same context already represented appropriately by our routing. This may not be consistently applied throughout the navigation helpers, and if not, we should visit that at some point and do it consistently one way or another. You'll see in some connectors we pass `routeParams` and in others we break down routeParams into the various keys an then basically reconstruct the routeParams object when we generate a URL. There are important exceptions as in the cases where we don't want to reproduce the whole context on a link on purpose. In those cases we simply pass the keys we want to use to reproduce the link. Hope that makes sense. I obviously like the idea of passing around `routeParams` but I'm explaining this all more to make clear the direction and intent of the existing design more than to close any conversation about it :)